### PR TITLE
feat: add renku headers to nb service requests

### PR DESCRIPTION
**Pull request by ableuler**
This PR adds the header
`Renku-Gitlab-Accesstoken: <some-token>`
together with some other Renku specific header to authenticated requests to the notebooks service. Note that this is only a temporary solution as the expected header names / content will be synchronized between the different renku backend services.

/deploy

### CODE REVIEW

File: helm-chart/renku-gateway/templates/configmap.yaml [hunk 1]
```diff
@@ -201,6 +201,11 @@ data:
           trustForwardHeader = true
           authResponseHeaders = ["Authorization", "Renku-user-id", "Renku-user-fullname", "Renku-user-email"]
 
+        [http.middlewares.auth-notebook.forwardauth]
+          address = "http://{{ template "gateway.fullname" . }}-auth/?auth=notebook"
+          trustForwardHeader = true
+          authResponseHeaders = ["Renku-gitlab-accesstoken", "Renku-user-id", "Renku-user-fullname", "Renku-user-email"]
```
Comment (olevski) [inline helm-chart/renku-gateway/templates/configmap.yaml:207]: @ableuler I see all of `"Renku-gitlab-accesstoken", "Renku-user-id", "Renku-user-fullname", "Renku-user-email"` in the header inside the requests in the notebooks service. However I do not see just the regular `Renku-User` header. Was this also supposed to be included here? If I include `Renku-User` in the list of headers here it does show up in notebooks.

**Review by olevski — CHANGES_REQUESTED** [review]
Andreas I just have one question and a few nit-picky things. Everything is in order though.

**Comment by ableuler** [comment]
Thanks @olevski and sorry for this rather ugly copy-paste PR. Let's keep the PR open so that the notebook service can be developed further against this branch and in the mean time figure out the proper headers structure we want adopt in the future. I've updated the [issue description](https://github.com/SwissDataScienceCenter/renku-gateway/issues/406#issue-864623751).

File: helm-chart/renku-gateway/templates/configmap.yaml (hunk 1 above)
Comment (ableuler) [reply helm-chart/renku-gateway/templates/configmap.yaml:207]: There's a config to determine which headers will actually taken over from the response of the middleware: https://github.com/SwissDataScienceCenter/renku-gateway/blob/429a4004fc91badabef6fbe83a5151426f6bc045/helm-chart/renku-gateway/templates/configmap.yaml#L207

Comment (ableuler) [reply helm-chart/renku-gateway/templates/configmap.yaml:207]: So you'd also have to add `Renku-User` in this list in order to receive it in the notebooks service.

File: helm-chart/renku-gateway/templates/configmap.yaml [hunk 2]
```diff
@@ -320,7 +325,7 @@ data:
         [http.routers.notebooksSecondPassRegular]
           priority=1003
           entryPoints = ["http"]
-          Middlewares = ["common", "notebooks"]
+          Middlewares = ["auth-notebook","common", "notebooks"]
```
[the remark this answers is not in the document]
Comment (ableuler) [reply helm-chart/renku-gateway/templates/configmap.yaml:328]: 👍

**Comment by ableuler** [comment]
I've now implemented the header structure suggested in #406.

**Comment by ableuler** [comment]
@olevski I think it would be good to merge this PR. It has worked with the operator-backed session POC and doesn't break any of the current functionality.

**Review by olevski — APPROVED** [review]

**Pull request merged**